From: =?utf-8?q?Lo=C3=AFc_Minier?= <lool@dooz.org>
Date: Sat, 21 Sep 2013 23:37:03 +0200
Subject: Don't query immodules for the local tree when building statically or
 cross-compiling

[mbiebl: follow upstream and use immodules.cache instead of gtk.immodules]

Bug: http://bugzilla.gnome.org/show_bug.cgi?id=346531
---
 configure.ac              |  1 +
 modules/input/Makefile.am | 17 ++++++++++++++++-
 2 files changed, 17 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index b75f9c6..1d36929 100644
--- a/configure.ac
+++ b/configure.ac
@@ -150,6 +150,7 @@ esac
 dnl Initialize libtool
 AC_PROG_CC
 AM_DISABLE_STATIC
+AM_CONDITIONAL([ENABLE_STATIC], [test "$enable_static" = yes])
 
 dnl 
 dnl Check for a working C++ compiler, but do not bail out, if none is found.
diff --git a/modules/input/Makefile.am b/modules/input/Makefile.am
index c8610fa..091ce70 100644
--- a/modules/input/Makefile.am
+++ b/modules/input/Makefile.am
@@ -189,8 +189,12 @@ dist_imconffile_DATA = im-multipress.conf
 if CROSS_COMPILING
 RUN_QUERY_IMMODULES_TEST=false
 else
+if ENABLE_STATIC
+RUN_QUERY_IMMODULES_TEST=false
+else
 RUN_QUERY_IMMODULES_TEST=test -z "$(DESTDIR)"
 endif
+endif
 
 # Running this if cross compiling or if DESTDIR is set is going to
 # not work at all, so skip it.
@@ -253,7 +257,18 @@ noinst_LTLIBRARIES =				\
 included-modules: $(noinst_LTLIBRARIES)
 
 immodules.cache: Makefile.am $(module_LTLIBRARIES)
-	$(top_builddir)/gtk/gtk-query-immodules-2.0 $(module_LTLIBRARIES) > immodules.cache
+	@if $(RUN_QUERY_IMMODULES_TEST) ; then \
+	  echo "$(top_builddir)/gtk/gtk-query-immodules-2.0 > immodules.cache" ; \
+	  $(top_builddir)/gtk/gtk-query-immodules-2.0 $(module_LTLIBRARIES) > immodules.cache; \
+	else \
+	  echo "***" ; \
+	  echo "*** Warning: immodules.cache not built" ; \
+	  echo "***" ; \
+	  echo "*** Generate this file manually on host" ; \
+	  echo "*** system using gtk-query-immodules-2.0" ; \
+	  echo "***" ; \
+	  touch immodules.cache; \
+        fi
 
 CLEANFILES = immodules.cache
 
